pkg/filters/args.go

```diff
@@ -34,18 +34,20 @@ func (filter *ArgFilter) Filter(eventID events.ID, args []trace.Argument) bool {
 	for argName, filter := range filter.filters[eventID] {
 		found := false
 		var argVal interface{}
+		var argType string
 		for _, arg := range args {
 			if arg.Name == argName {
 				found = true
 				argVal = arg.Value
+				argType = arg.Type
 				break
 			}
 		}
 		if !found {
 			return true
 		}
-		// TODO: use type assertion instead of string conversion
-		if argName != "syscall" {
+		// TODO: parse to correct filter type per arg type
+		if argName != "syscall" && argType != "struct sockaddr*" && argType != "const struct sockaddr*" {
```

Comment:
NIT: I would go for something like:

```go
		switch argName {
		case "syscall":
		default:
			switch argType {
			case "struct sockaddr*", "const struct sockaddr*":
			default:
				argVal = fmt.Sprint(argVal)
		}
```

because of "patchability" (this way patches adding/removing types here won't change the entire line, causing merge conflicts when managing the code).

Reply:
Just saw you did something similar to what I proposed bellow (in Parse()). I would keep both the way I suggested so we can extend special cases arg names ("syscall" like) and special cases arg types, just adding them to switch cases.